### PR TITLE
Add support for release bundles for permission targets

### LINKED ISSUE #449: artifactory_permission_target does not support release_bundles
**Is your feature request related to a problem? Please describe.**
I cannot use terraform to configure permission targets which manage release bundles because the terraform provider does not support it.  To make matters more difficult, the schema implemented within the provider does support release_bundles but this configuration is silently ignored when applying the resource state causing inconsistency between the state and the resource configuration.

**Describe the solution you'd like**
I would like to be able to configure permission targets which manage permissions for release bundles.  I plan to put up a PR shortly which adds this support, so really I would like a review and collaboration to get it integrated.

**Describe alternatives you've considered**
I need to manually update the target either using  the Rest API or the UI.

**Additional context**
I plan to put up a PR to the provider when I have time to implement it, but I wanted to get the issue filed so I could document it in my terraform code in the interim.

### PR DESCRIPTION
**Pull request by neilvana**
To resolve #449, add support for release bundles to `artifactory_permission_target`.  Previously the schema already supported specifying a `release_bundle` block but the pack and unpack methods lacked the capability to operate on the block.  Implement the required logic, add documentation and augment test cases.

I tested I was able to create a new permission target including release bundle permissions on our Enterprise+ SaaS instance of Artifactory.  I have not yet run the acceptance tests since I have yet to set up a test Artifactory server. Would it be possible to run these tests on JFrog's infrastructure or do I need to set up my own trial instances of Artifactory to validate the tests?

### CODE REVIEW

**Comment by alexhung** [comment]
@neilvana Thanks for the PR! The PR will be run through our CI system so we will know soon.

**Comment by alexhung** [comment]
@chb0github ☝🏼

**Comment by neilvana** [comment]
I'm working on getting the CLA signed.

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go [hunk 1]
```diff
@@ -51,6 +51,27 @@ const permissionJustBuild = `
 	}
 `
 
+const permissionJustReleaseBundle = `
+	//resource "artifactory_local_docker_repository" "{{ .repo_name }}" {
```
Comment (chb0github) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:55]: Remove commented out code

**Comment by martinm82** [comment]
Thanks a lot for this PR. The perfect moment as we started to use recently the provider and release bundles 💯

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go (hunk 1 above)
Comment (neilvana) [reply pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go]: Done.

**Comment by alexhung** [comment]
@neilvana Got the test results from our CI. I'll look into this and see if I can fix this on my end and get the tests pass locally.

```
=== RUN   TestAccPermissionTarget_addBuild
    resource_artifactory_permission_target_test.go:243: Step 4/4 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 49, in resource "artifactory_permission_target" "test-perm199069":
          49: 		excludes_pattern = ["bar/**]
          50: 		repositories     = ["release-bundles"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Unterminated template string
        
          on terraform_plugin_test.tf line 49, in resource "artifactory_permission_target" "test-perm199069":
          49: 		excludes_pattern = ["bar/**]
          50: 		repositories     = ["release-bundles"]
        
        No closing marker was found for the string.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 50, in resource "artifactory_permission_target" "test-perm199069":
          50: 		repositories     = ["release-bundles"]
          51: 		actions {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 52, in resource "artifactory_permission_target" "test-perm199069":
          52: 		actions {
          53: 			users {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 53, in resource "artifactory_permission_target" "test-perm199069":
          53: 			users {
          54: 				name        = "anonymous"
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 54, in resource "artifactory_permission_target" "test-perm199069":
          54: 				name        = "anonymous"
          55: 				permissions = ["read", "write"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 55, in resource "artifactory_permission_target" "test-perm199069":
          55: 				permissions = ["read", "write"]
          56: 			}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 56, in resource "artifactory_permission_target" "test-perm199069":
          56: 			}
          57: 			groups {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 59, in resource "artifactory_permission_target" "test-perm199069":
          59: 			groups {
          60: 				name        = "readers"
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 60, in resource "artifactory_permission_target" "test-perm199069":
          60: 				name        = "readers"
          61: 				permissions = ["read"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 61, in resource "artifactory_permission_target" "test-perm199069":
          61: 				permissions = ["read"]
          62: 			}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 62, in resource "artifactory_permission_target" "test-perm199069":
          62: 			}
          63: 		}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 63, in resource "artifactory_permission_target" "test-perm199069":
          63: 		}
          64: 	  }
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 64, in resource "artifactory_permission_target" "test-perm199069":
          64: 	  }
          65:      //depends_on = [artifactory_local_docker_repository.example-repo-local]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 65, in resource "artifactory_permission_target" "test-perm199069":
          65:      //depends_on = [artifactory_local_docker_repository.example-repo-local]
          66: 	}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 66, in resource "artifactory_permission_target" "test-perm199069":
          66: 	}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
    testing_new.go:72: Error retrieving state, there may be dangling resources: exit status 1
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 49, in resource "artifactory_permission_target" "test-perm199069":
          49: 		excludes_pattern = ["bar/**]
          50: 		repositories     = ["release-bundles"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Unterminated template string
        
          on terraform_plugin_test.tf line 49, in resource "artifactory_permission_target" "test-perm199069":
          49: 		excludes_pattern = ["bar/**]
          50: 		repositories     = ["release-bundles"]
        
        No closing marker was found for the string.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 50, in resource "artifactory_permission_target" "test-perm199069":
          50: 		repositories     = ["release-bundles"]
          51: 		actions {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 52, in resource "artifactory_permission_target" "test-perm199069":
          52: 		actions {
          53: 			users {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 53, in resource "artifactory_permission_target" "test-perm199069":
          53: 			users {
          54: 				name        = "anonymous"
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 54, in resource "artifactory_permission_target" "test-perm199069":
          54: 				name        = "anonymous"
          55: 				permissions = ["read", "write"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 55, in resource "artifactory_permission_target" "test-perm199069":
          55: 				permissions = ["read", "write"]
          56: 			}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 56, in resource "artifactory_permission_target" "test-perm199069":
          56: 			}
          57: 			groups {
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 59, in resource "artifactory_permission_target" "test-perm199069":
          59: 			groups {
          60: 				name        = "readers"
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 60, in resource "artifactory_permission_target" "test-perm199069":
          60: 				name        = "readers"
          61: 				permissions = ["read"]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 61, in resource "artifactory_permission_target" "test-perm199069":
          61: 				permissions = ["read"]
          62: 			}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 62, in resource "artifactory_permission_target" "test-perm199069":
          62: 			}
          63: 		}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 63, in resource "artifactory_permission_target" "test-perm199069":
          63: 		}
          64: 	  }
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 64, in resource "artifactory_permission_target" "test-perm199069":
          64: 	  }
          65:      //depends_on = [artifactory_local_docker_repository.example-repo-local]
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 65, in resource "artifactory_permission_target" "test-perm199069":
          65:      //depends_on = [artifactory_local_docker_repository.example-repo-local]
          66: 	}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
        
        Error: Invalid multi-line string
        
          on terraform_plugin_test.tf line 66, in resource "artifactory_permission_target" "test-perm199069":
          66: 	}
        
        Quoted strings may not be split over multiple lines. To produce a multi-line
        string, either use the \n escape to represent a newline character or use the
        "heredoc" multi-line template syntax.
        
--- FAIL: TestAccPermissionTarget_addBuild (4.71s)
```

**Review by alexhung — CHANGES_REQUESTED** [review]
Tests failed

**Review by chb0github — DISMISSED** [review]
I tests are passing, @alexhung I am ok with this

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ const permissionFull = `
 
 	  release_bundle {
 		includes_pattern = ["foo/**"]
-		excludes_pattern = ["bar/**]
+		excludes_pattern = ["bar/**"]
```
Comment (chb0github) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:118]: Oopa

**Comment by neilvana** [comment]
I missed a quote in the test template 🤦  It should be fixed now.  Would it be possible to run the CI tests again?

**Comment by alexhung** [comment]
@neilvana Tests pass in CI.

**Review by alexhung — APPROVED** [review]
:shipit:

**Review by chb0github — APPROVED** [review]

**Comment by alexhung** [comment]
I'll merge this after #456 and #453

**Comment by alexhung** [comment]
@neilvana Any progress on signing the CLA?

**Comment by neilvana** [comment]
Hopefully by tomorrow.  I'm waiting on Legal.  Since this is the first contribution from our organization to JFrog repos it is taking a bit longer.  Sorry for the delay.  I'll try to hurry it up.

**Review by danielmkn — APPROVED** [review]

**Comment by neilvana** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**